src/features/sarcophagi/components/Rewrap.tsx

```diff
@@ -2,16 +2,23 @@ import { Button, HStack, Text, VStack } from '@chakra-ui/react';
 import { DatePicker } from 'components/DatePicker';
 import { DatePickerButton } from 'components/DatePicker/DatePickerButton';
 import { BigNumber, ethers } from 'ethers';
+import { formatEther } from 'ethers/lib/utils';
 import { useRewrapSarcophagus } from 'hooks/embalmerFacet';
-import { useGetSarcophagusDetails } from 'hooks/viewStateFacet';
+import { useGetProtocolFeeAmount, useGetSarcophagusDetails } from 'hooks/viewStateFacet';
+import { useGetSarcophagusArchaeologists } from 'hooks/viewStateFacet/useGetSarcophagusArchaeologists';
 import { buildResurrectionDateString } from 'lib/utils/helpers';
-import { useState } from 'react';
+import { useMemo, useState } from 'react';
 import { useNavigate, useParams } from 'react-router-dom';
 
 export function Rewrap() {
   const { id } = useParams();
   const navigate = useNavigate();
   const { sarcophagus } = useGetSarcophagusDetails({ sarcoId: id });
+  const archaeologists = useGetSarcophagusArchaeologists(
+    id || ethers.constants.HashZero,
+    sarcophagus?.archaeologists
+  );
+  const protocolFeeAmountInt = useGetProtocolFeeAmount();
```

Comment:
This seems like a good candidate for caching. There would be no need to hit the network. We can make a task to look into caching which I think it available on wagmi.